lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4298,19 +4298,21 @@ namespace {
         }
       }
 
-      if (!found) {
-        // Go back to the first list and find classes with matching Swift names
-        // *even if the ObjC name doesn't match.*
-        // This shouldn't be allowed but we need it for source compatibility;
-        // people used `@class SwiftNameOfClass` as a workaround for not
-        // having the previous loop, and it "worked".
-        for (auto result : swiftDeclsByName) {
-          if (auto singleResult = dyn_cast<T>(result)) {
-            if (isMatch(singleResult, /*baseNameMatches=*/true,
-                        /*allowObjCMismatch=*/true)) {
-              if (found)
-                return nullptr;
-              found = singleResult;
+      if (!languageVersion.isVersionAtLeast(6)) {
```

Comment:
This fallback was kept around so as to not break source compatibility, but it was never intended to work and should probably stop working in the next major version: https://github.com/apple/swift/pull/27921